### PR TITLE
Fix bug with multi quests

### LINKED ISSUE #2804: [BUG] Cannot turn in completed Ultra Beast Hunt quest with Buzzwole and Pheromosa
**Info**
 - OS: Windows 10
 - Browser : Chrome
 - Version : 0.9.10

**Description:**

The steps for Ultra Beast Hunt quest with Buzzwole and Pheromosa are completed, but I cannot proceed in the quest line. I did catch them with beast balls, and they are both 2/2 and 4/4 in green. I'm not using scripts/autoclicker.

**Steps to Reproduce:**

1. Talked to 
2. Captured 2 Buzzwole with Beast balls
3. Captued 4 Pheromosa with Beast balls
4. Talked to Anabel : she tells me to go catch them
5. Reloaded the game
6. Talked to Anabel again : same thing
7. Captured one more Buzzwole and Pheromosa
8. Talked to Anabel again : same thing

**Expected Behavior:**

Quest should proceed to the next step

**Screenshots:**

image
image
image
image

**Additional Information:**

 I started the quest on 0.9.9 or a little earlier. The issue is still there on 0.9.10, and importing my 0.9.9 backup save from today does not fix this (meaning it was propably not introduced  in 0.9.10).

### PR DESCRIPTION
**Pull request by Jaaslet**
Fixes #2804

### CODE REVIEW

File: src/scripts/quests/Quests.ts [hunk 1]
```diff
@@ -305,7 +305,7 @@ class Quests implements Saveable {
                 if (ql) {
                     ql.state(questLine.state);
                     if (questLine.state == QuestLineState.started) {
-                        ql.resumeAt(questLine.quest, questLine.initial);
+                        ql.resumeAt(questLine.quest, (questLine.initial instanceof Number ? questLine.initial : 0));
```
Comment (RedSparr0w) [inline src/scripts/quests/Quests.ts:308]: I think this should be returning undefined, as if they already have progress in the statistic it's tracking, the quest will just auto complete.
```suggestion
                        ql.resumeAt(questLine.quest, (questLine.initial instanceof Number ? questLine.initial : undefined));
```

**Comment by Jaaslet** [comment]
Your change works, so it's submitted. I think it can just be merged, as your change is the only one since you looked at it?

**Comment by amative1** [comment]
Just leaving this comment here so this doesn't get pulled while I'm debugging. We've identified something fishy going on with this in the Meltan branch.

**Review by RedSparr0w — APPROVED** [review]
Looks good, thanks for the PR! 👍

**Comment by RedSparr0w** [comment]
> Just leaving this comment here so this doesn't get pulled while I'm debugging. We've identified something fishy going on with this in the Meltan branch.

This PR shouldn't be a part of another branch yet as it's not in dev?

**Comment by amative1** [comment]
I copied the code over to Meltan since it uses Multiquest, and we're getting very weird behavior

**Pull request closed without merging**